### DISCUSSION TITLE
Организация демок

### DISCUSSION THREAD

**Discussion by furtivite** [doka-guide/content#1343] [Q&A] [answered]
Сейчас у нас структура, где картинки, видео и так далее лежат на уровне статьи, например:

```
article/
  images/
    article.png
    demo-1.png
    demo-2.png
  scripts/
    demo-1.js
  demos/
    demo-1.html
    demo-2.html
  index.md
```

Все ресурсы демок в таком случае находятся на уровне статьи. Какие у этого проблемы:

1. Обращаться к ресурсам приходится на уровень выше `../images/picture.png`
2. Непонятно, какие ресурсы относятся только к демке, а какие только к статье
3. Если делать демку локально, то при добавлении придётся перемещать ресурсы и менять пути
4. Если хочется скопировать демку (и сделать на основе другую), непонятно что копировать

## Решение №1

Иметь в демках свою папку с ресурсами:

```
article/
  images/
    article.png
  demos/
    images/
      demo-1.png
      demo-2.png
    scripts/
      demo-1.js
    demo-1.html
    demo-2.html
  index.md
```

Так станет понятно, что ресурсы только для демок. Для всей группы демок.

## Решение №2

Изолировать все демки в своих папках

```
article/
  images/
    article.png
  demos/
    demo-1/
      images/
        demo-1.png
      scripts/
        demo-1.js
      index.html
    demo-2/
      images/
        demo-2.png
      index.html
  index.md
```

Так станет понятно, что демка — это папка и всё внутри неё только её.

**Comment by pepelsbey** [comment]
@furtivite обновил описание

**Comment by nlopin** [comment]
Если менять структуру, то вторая кажется максимально логичной

**Comment by skorobaeus** [comment]
Имхо, первая структура немного лучше подходит, потому что ресурсы в демках одной статьи нередко реиспользуются: одна и та же библиотечка, одна и та же картинка и т.д.

Скромно надеюсь, что текущую структуру на выбранную можно будет поменять как-то автоматически, а не вручную.

**Reply by pepelsbey** [reply]
Мне кажется, что если у нас серия демок, использующих одни и те же ресурсы, то их можно положить в одну папку. То есть совместить плюсы 1 и 2 способов.

**Reply by igsekor** [reply]
Давайте `index.html` держать для каждой демки. Пожалуйста 🙏🏻

**Reply by nlopin** [reply]
Я считаю, что переиспользование создаёт сложность в организации. Появляются дополнительные правила что и куда складывать, искать потом становится сложнее. 

Давайте сделаем дублирование и если его получится очень много, то отдельной итерацией улучшим

**Comment by igsekor** [comment]
А ещё бывают демо в практиках. Кажется, что это оверкил, надо все демки класть в папку `demos`, которая в папке статьи. Что думаете?

**Reply by nlopin** [reply]
Согласен, давайте все демки в одном месте держать

**Comment by pepelsbey** [comment, marked as answer]
[Сделал пул-реквест](https://github.com/doka-guide/content/pull/1377) с обновлённой документацией для демок, которая описывает вариант №2: у каждой демки своя папка.
